flink-streaming-connectors/flink-connector-elasticsearch2/src/main/java/org/apache/flink/streaming/connectors/elasticsearch2/ElasticsearchSink.java

```diff
@@ -186,22 +191,44 @@ public void beforeBulk(long executionId, BulkRequest request) {
 
 			@Override
 			public void afterBulk(long executionId, BulkRequest request, BulkResponse response) {
+				boolean allRequestsRepeatable = true;
 				if (response.hasFailures()) {
 					for (BulkItemResponse itemResp : response.getItems()) {
 						if (itemResp.isFailed()) {
-							LOG.error("Failed to index document in Elasticsearch: " + itemResp.getFailureMessage());
-							failureThrowable.compareAndSet(null, new RuntimeException(itemResp.getFailureMessage()));
+							// Check if index request can be retried
+							String failureMessageLowercase = itemResp.getFailureMessage().toLowerCase();
+							if (failureMessageLowercase.contains("timeout") || failureMessageLowercase.contains("timed out") // Generic timeout errors
```

Comment:
I didn't find an alternative to check strings. I will add a flag an disable it by default.